calico/openstack/test/test_plugin_etcd.py

```diff
@@ -129,7 +152,15 @@ def etcd_read(self, key, recursive=False, timeout=None):
             print "children: %s" % [child.key
                                     for child in read_result.children]
             if read_result.value is None and read_result.children == []:
-                raise lib.m_etcd.EtcdKeyNotFound()
+                raise lib.m_etcd.EtcdKeyNotFound(self.etcd_data)
+
+            read_result._children = []##actual direct children of the dir in etcd response. needed for status_dir, where children are dirs and gotta be iterated
```

Comment:
As above, comment should be prefixed with a single hash. Since this is a long comment, it should probably be on its own line and wrapped to 72 characters.